src/views/TimezoneModal.vue

```diff
@@ -0,0 +1,167 @@
+    </ion-toolbar>
+  </ion-header>
+
+  <ion-content class="ion-padding">
+    <!-- Empty state -->
+    <div class="empty-state" v-if="filteredTimeZones.length === 0">
+      <p>{{ $t("No time zone found")}}</p>
+    </div>
+
+    <!-- Timezones -->
+    <div v-else>
+      <ion-list>
+        <ion-radio-group value="rd" v-model="timeZoneId">
+          <ion-item :key="timeZone.id" v-for="timeZone in filteredTimeZones">
+            <ion-label>{{ timeZone.label }} ({{ timeZone.id }})</ion-label>
+            <ion-radio :value="timeZone.id" slot="start" />
+          </ion-item>
+        </ion-radio-group>
+      </ion-list>
+    </div>
+    
+    <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+      <ion-fab-button :disabled="!timeZoneId" @click="saveAlert">
+        <ion-icon :icon="save" />
+      </ion-fab-button>
+    </ion-fab>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import { 
+  IonButtons,
+  IonButton,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonItem,
+  IonIcon,
+  IonLabel,
+  IonList,
+  IonRadioGroup,
+  IonRadio,
+  IonSearchbar,
+  IonTitle,
+  IonToolbar,
+  modalController,
+  alertController } from "@ionic/vue";
+import { defineComponent } from "vue";
+import { close, save } from "ionicons/icons";
+import { useStore } from "@/store";
+import { UserService } from "@/services/UserService";
+import { hasError } from '@/utils'
+import { DateTime } from 'luxon';
+
+export default defineComponent({
+  name: "TimeZoneModal",
+  components: { 
+    IonButtons,
+    IonButton,
+    IonContent,
+    IonFab,
+    IonFabButton,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonRadioGroup,
+    IonRadio,
+    IonSearchbar,
+    IonTitle,
+    IonToolbar 
+  },
+  data() {
+    return {
+      queryString: '',
+      filteredTimeZones: [],
+      timeZones: [],
+      timeZoneId: ''
+    }
+  },
+  methods: {
+    closeModal() {
+      modalController.dismiss({ dismissed: true });
+    },
+    async saveAlert() {
+      const message = this.$t("Are you sure you want to change the time zone to?", { timeZoneId: this.timeZoneId });
+      const alert = await alertController.create({
+        header: this.$t("Update time zone"),
+        message,
+        buttons: [
+          {
+            text: this.$t("Cancel"),
+          },
+          {
+            text: this.$t("Confirm"),
+            handler: () => {
+              this.setUserTimeZone();
+            }
+          }
+        ],
+      });
+      return alert.present();
+    },
+    escapeRegExp(text: string) {
+      //TODO Handle it in a better way
+      // Currently when the user types special character as it part of Regex expressions it breaks the code
+      // so removed the characters for now
+      return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+    },
+    findTimeZone() { 
+      const regularExp = new RegExp(`${this.escapeRegExp(this.queryString)}`, 'i');
```

Comment:
We could disallow user to type such characters with keyup/keydown events

Reply:
Done sir